app/models/lookup_value.rb

```diff
@@ -98,6 +98,18 @@ def ensure_matcher(match_type)
     matcher
   end
 
+  def ensure_matcher_exists
+    return false if match.blank?
+    match.split(LookupKey::KEY_DELM).each do |m|
+      key_element = m.split(LookupKey::EQ_DELM).first
+      unless lookup_key.path.include?(key_element)
```

Comment:
I think `path` needs to be split here to more precisely check whether the attribute is in the _array_ of path items, not just the path _string_ as a whole. The path array should be accessible through `path_elements`.

I think this is why some tests that used single quotes for `path` are passing when they shouldn't (see comments further down).